plutus-core/plutus-ir/Language/PlutusIR/Compiler/Types.hs

```diff
@@ -51,16 +51,16 @@ defaultCompilationOpts :: CompilationOpts
 defaultCompilationOpts = CompilationOpts True
 
 data CompilationCtx uni a = CompilationCtx {
-    _ccOpts        :: CompilationOpts
+    _ccOpts              :: CompilationOpts
     , _ccBuiltinMeanings :: PLC.DynamicBuiltinNameMeanings (PIR.Term PLC.TyName PLC.Name uni ())
-    , _ccEnclosing :: Provenance a
-    , _ccTypeCheckConfig :: PirTCConfig uni
+    , _ccEnclosing       :: Provenance a
+    , _ccTypeCheckConfig :: Maybe (PirTCConfig uni)
```

Comment:
This now does two things, both telling us whether to typecheck and if so how to do it. Worth a comment at least.